### PR TITLE
fix: panic at empty http spec

### PR DESCRIPTION
**Pull request by basuotian**
Signed-off-by: Basuotian <basuoluomiu@gmail.com>

### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### What this PR does / why we need it:

Fix: #1659

### Pre-submission checklist:

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by basuotian** [comment]
@tao12345666333 PTAL, thx

**Comment by Fabriceli** [comment]
i think this #1654 is the same situation

**Review by tao12345666333 — COMMENTED** [review]
Thanks

Can you create an e2e test case to cover this?

**Comment by basuotian** [comment]
@tao12345666333 add an e2e-test case for empty http spec, but i am not sure the case is expected, PTAL, thx

File: test/e2e/suite-ingress/suite-ingress-resource/ingress.go [hunk 1]
```diff
@@ -466,6 +466,39 @@ spec:
...
+              number: %d
+`, backendSvc, backendPort[0])
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ing), "creating ingress")
+		assert.Nil(ginkgo.GinkgoT(), s.EnsureNumApisixRoutesCreated(1))
+
+		_ = s.NewAPISIXClient().GET("/status/500").WithHeader("Host", "httpbin.org").Expect().Status(http.StatusInternalServerError)
+		_ = s.NewAPISIXClient().GET("/status/504").WithHeader("Host", "httpbin.org").Expect().Status(http.StatusGatewayTimeout)
+		_ = s.NewAPISIXClient().GET("/statusaaa").WithHeader("Host", "httpbin.org").Expect().Status(http.StatusNotFound).Body().Contains("404 Route Not Found")
+		// Mismatched host
+		_ = s.NewAPISIXClient().GET("/status/200").WithHeader("Host", "a.httpbin.org").Expect().Status(http.StatusNotFound)
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-resource/ingress.go:500]: Using empty.httpbin.org

Comment (basuotian) [reply test/e2e/suite-ingress/suite-ingress-resource/ingress.go]: done, empty.httpbin.org for 404 status

**Review by tao12345666333 — APPROVED** [review]

**Comment by basuotian** [comment]
@AlinsRan  first pr need one more reviewer, PTAL, thx

**Review by AlinsRan — APPROVED** [review]
